src/hotspot/share/gc/g1/g1FullGCMarker.hpp

```diff
@@ -96,6 +99,12 @@ class G1FullGCMarker : public CHeapObj<mtGC> {
   CLDToOopClosure*      cld_closure()   { return &_cld_closure; }
   G1MarkAndPushClosure* mark_closure()  { return &_mark_closure; }
   G1FollowStackClosure* stack_closure() { return &_stack_closure; }
+
+  void flush_mark_region_cache() {
+    if (MarkSweepDeadRatio > 0) {
```

Comment:
Drop this check to make the code more straightforward. Other code like in PR #2579 might find this information useful too. Maybe this could even be factored out in a separate CR.